Lib/importlib/_bootstrap_external.py

```
@@ -104,28 +104,29 @@ def _path_join(*path_parts):
             return ""
         if len(path_parts) == 1:
             return path_parts[0]
-        root = ""
+        anchor = ""
         path = []
-        for new_root, tail in map(_os._path_splitroot, path_parts):
-            if new_root.startswith(path_sep_tuple) or new_root.endswith(path_sep_tuple):
-                root = new_root.rstrip(path_separators) or root
+        for drive, root, tail in map(_os._path_splitroot, path_parts):
```

Previously this could handle `Path` objects (and people tried it by adding them to `sys.path`, which has a ~50/50 chance of ever working). Now it will fail early enough to reduce that to a 0% chance of working. IIRC, we don't have `os.fsdecode` here, but using `path_t` in the native function works correctly.

I'm not totally against that, but it's worth understanding the situations this will impact and making sure they're correctly documented.

---

I've restored the old implementation and renamed it to `_path_splitanchor`. Let's leave removing it for another time.

---

It's literally just the use of `path_t` instead of `unicode` in the new definition. I know that adds an extra few checks in C code to covert back to bytes, but that saves the same check currently in Python code and it's really not that difficult (you'll see it in most other functions already).

But if we're going to leave the old one there, leave it with its original name and make the new function a new function.

---

I think I figured it out, but now I'm getting a segmentation fault on Unix.